Api-docs for profile: get - /users/:userId/profile

VerumEmanio opened the pull request:
@MambaDev I used your sample response from #28 to document the findOne endpoint.

Questions:
1. Can you provide me with the endpoint URL? I don't know where to find it.
2. What changes should I make?
3. What branch should I be making pull requests to? I only made it to master because there isn't a branch for api documentation, and I'm unable to create a new branch on your repo.

MambaDev commented on app/controllers/user/UserProfile.controller.ts:
```diff
@@ -5,6 +5,62 @@ import UserProfile from '../../models/UserProfile';
 
+/**
+ * @api {get} NEED ENDPOINT URL Shows all of the profile information for a certain username.
+ * @apiName ShowProfileforUsername
```
`Show` can probably be replaced to `Get`, and `Username` to `User` (for more generic reading) since all users endpoints are based on the id of the user and not the username.

MambaDev commented on app/controllers/user/UserProfile.controller.ts:
```diff
@@ -5,6 +5,62 @@ import UserProfile from '../../models/UserProfile';
 import { IUserRequest } from '../../request/IRequest';
 
+/**
+ * @api {get} NEED ENDPOINT URL Shows all of the profile information for a certain username.
```
/users/:userId/profile

MambaDev commented:
Hi, @VerumEmanio,

This looks great so far 🎉. I will try and answer some questions for you and provide some feedback.

> Can you provide me with the endpoint URL? I don't know where to find it.

So the endpoint addresses are built from the base up. This starts within the index.ts file within the routes directory. `/app/routes/index.ts`. The root address will then lead into an area-specific route and then to the function.

For the show method you have documented, this would be the following: `index.ts -> User.routes.ts` and the endpoint address would be `/users/:userId/profile`.

> What branch should I be making pull requests to? 

Requesting a pull request from your fork into master is fine for now since the DevWars repository does not have a Development/Staging branches. Some best practice advice would be to work on a separate branch on your fork and merge that in. This allows you to update your fork from our master without causing any problems.

### Feedback
* ShowProfileforUsername
The name of the API and the way the endpoint is discussed is talked in the terms of "show", I know the function name is misleading but it would be referred to match the other docs with something like GetProfileForUser

* Username
The user endpoint you have mentioned talks in the sense of username. The user routes are based on the id of the user and not the username. So in cases in which you have used username, just use user or user by id.

* Be sure to join our discord!  https://discord.gg/devwars

VerumEmanio commented:
Thanks, I'm glad you like it! I updated the file to reflect your wording changes.

I'm still a bit confused about how to determine the endpoint address. Here's what I'm doing based on your instructions for this endpoint:
1. I open `/app/routes/index.ts`.
2. I see that line 10 `import { UserRoute } from './User.routes';` tells me to open `User.routes.ts`. Before I open that, I also get the beginning of the address from line 64 `path: '/users',`.
3. I open  `/app/routes/User.routes.ts`. This is where I get stuck because I don't see `/:userId/profile` anywhere.

MambaDev commented:
@VerumEmanio,

we can see [here](https://github.com/DevWars/devwars-api/blob/4c6bf6973c630428cccdf641fa35324793ee4433/app/routes/User.routes.ts#L67) that we have a profile section for User.routes.ts. Here it contains two requests for profile. A get and patch. The first parameter is the url after /users/.

Hope this helps.

MambaDev commented on app/controllers/user/UserProfile.controller.ts:
```diff
@@ -5,6 +5,62 @@ import UserProfile from '../../models/UserProfile';
 
+/**
+ * @api {get} /users/:userId/profile Get all of the profile information for a certain user.
+ * @apiName ShowProfileforUser
```
GetProfileForUser as per the feedback.

MambaDev commented on app/controllers/user/UserProfile.controller.ts:
```diff
@@ -5,6 +5,62 @@ import UserProfile from '../../models/UserProfile';
+ * @apiName ShowProfileforUser
+ * @apiGroup User
+ * 
+ * @apiParam {string} username A partial or full username for a given user.
```
id, the id is used as the parameter within the given url. Never usernames since these can change.

VerumEmanio commented:
I updated the apiName and added permissions info. Thanks for thoroughly explaining things to me and for providing links so I can learn more. I'm excited to contribute more documentation! I'll try to document the second endpoint in this file.

Is the inline documentation outputted in its polished form somewhere that I can access? It'd be cool to see it outside of Visual Studio Code.

MambaDev commented:
Hi @VerumEmanio 

Here is a snippet for your docs (you can also see it locally with : `npm run serve-docs`):

![image](https://user-images.githubusercontent.com/12329422/72164873-689ace00-33be-11ea-89e4-742e4c7b008f.png)
